static/js/containers/UserPage_test.js

```diff
@@ -77,13 +78,64 @@ describe("UserPage", function() {
           getElementsByClassName('delete-button')[0];
       };
 
+      beforeEach(() => {
+        let userProfile = Object.assign({}, USER_PROFILE_RESPONSE, {
+          username: SETTINGS.username
+        });
+        userProfile.education.push({
+          "id": 3,
+          "degree_name": DOCTORATE,
+          "graduation_date": "2015-12-01",
+          "field_of_study": "Philosophy",
+          "school_name": "Harvard",
+          "school_city": "Cambridge",
+          "school_state_or_territory": "US-MA",
+          "school_country": "US",
+          "online_degree": false
+        });
+        helper.profileGetStub.
+          withArgs(SETTINGS.username).
+          returns(Promise.resolve(userProfile));
+      });
+
       it('shows the education component', () => {
         return renderComponent(`/users/${SETTINGS.username}`, userActions).then(([, div]) => {
           let title = div.getElementsByClassName('profile-card-title')[1];
           assert.equal(title.textContent, 'Education');
         });
       });
 
+      it('should show the entries in resume order', () => {
```

Comment:
yeah the `then` handler function can be refactored to something like this:

```javascript
        let testDialogIndexStuff = (sortedEntries, unsortedEntries, dialogIndexKey) => {
          state = helper.store.getState();
          let stateIndex = state.ui[dialogIndexKey];
          let [sortIndex, sortEntry] = sortedEntries[0];
          // the work dialog index dispatched to the store should be the same index
          // as the index (into the unsorted list) of the first element in our sorted list
          // since we clicked on the first work entry in the UI
          assert.equal(stateIndex, sortIndex);
          // this index should not be equal to 0, since the first element in the sorted list
          // should not be the first item in the unsorted list
          assert.notEqual(stateIndex, 0);

          // the entry the index in the state points to should be the same element 
          // in the unsorted education that we have in the sorted array\
          assert.deepEqual(sortEntry, unsortedEntries[stateIndex]);
        };
```

You could throw the `assert.notDeepEqual` stuff in there too if you wanted. the body of `renderComponent` would then look something like this:

```javascript
let state = helper.store.getState();
let unsortedEntries = state.profiles[SETTINGS.username].profile.education;
let sortedEntries = educationEntriesByDate(unsortedEntries);
let dialogActionHandler = _.partial(testDialogIndexStuff, sortedEntries, unsortedEntries, 'educationDialogIndex')

return listenForActions([
  SET_EDUCATION_DIALOG_INDEX,
  SET_EDUCATION_DIALOG_VISIBILITY,
  SET_EDUCATION_DEGREE_LEVEL,
], () => {
  TestUtils.Simulate.click(editButton);
}).then(dialogActionHandler);

```

maybe a bit quirky with the `_.partial`. you could also wrap the entire `listenForActions` bit and pass in the expected actions list in addition to the other arguments. either way, i think those are definitely preferable to two distinct test cases that test for the same type of functionality and are essentially copy/pasted